lib/reservation.rb

```diff
@@ -0,0 +1,45 @@
+require 'date'
+
+class Reservation
+
+  attr_reader :checkin_date, :checkout_date, :price, :room_number, :reservation_id, :customer
+
+  def initialize(checkin_date, checkout_date, room, customer, price_per_night = 200)
+    check_date_range(checkin_date, checkout_date)
+    @checkin_date = checkin_date
+    @checkout_date = checkout_date
+    @customer = customer
+    @room = room
+    @room_number = room.room_number
+    @reservation_id = create_unique_id(room_number, checkin_date)
+    @price_per_night = price_per_night
+    @price = calculate_price(checkin_date, checkout_date, price_per_night).to_i
+  end
+
+  def check_date_range(start, finish)
+    raise StandardError, 'Invalid date range' unless start < finish
+  end
+
+  def create_unique_id(room_number, checkin_date)
+    "#{room_number}000#{checkin_date.ld}".to_i
```

Comment:
How is this guaranteed to be unique?  It's likely to, but won't always be unique.

Also why have a method to get a Unique ID?  What purpose does it serve?